src/screens/ContactScreen/index.js

```diff
@@ -128,35 +131,36 @@ const ContactScreen = ({navigation}) => {
           })) || [];
 
         const dataUser = [...followedUsers, ...unfollowedUsers];
-        setUsers(
+        setUsersSearch(
           dataUser?.filter((item) =>
             isAddParticipant ? !existParticipants.includes(item.username) : item
           )
         );
       }
-      setLoading(false);
+      setIsLoadingSearchResult(false);
     } catch (error) {
-      setLoading(false);
+      setIsLoadingSearchResult(false);
     }
   };
 
   const debounced = React.useCallback(
     debounce((changedText) => {
-      handleSearch();
-      setDebouncedText(changedText);
+      handleSearch(changedText);
     }, 1000),
     []
   );
 
   React.useEffect(() => {
-    try {
-      setLoading(true);
-      getDiscoveryUser();
-      setLoading(false);
-    } catch (error) {
-      setLoading(false);
+    if (text.length === 0) {
+      try {
+        setLoading(true);
+        getDiscoveryUser();
+        setLoading(false);
+      } catch (error) {
+        setLoading(false);
+      }
     }
-  }, []);
+  }, [text]);
 
   React.useEffect(() => {
     if (users.length > 0) {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

> Note: This review was outside of the patch, so it was mapped to the patch with the greatest overlap. Original lines [115-143]

The `handleSearch` function has been updated to set the search results in `usersSearch` instead of `users`. This is a good change as it separates the search functionality from the regular user list, making the code easier to understand and maintain. However, there's a potential issue here: if the `DiscoveryRepo.fetchDiscoveryDataUser` function throws an error, `setIsLoadingSearchResult(false)` will not be called, leaving the UI in a loading state indefinitely. To fix this, you should use a finally block to ensure `setIsLoadingSearchResult(false)` is always called.

```diff
  const handleSearch = async (searchText) => {
    try {
      setIsLoadingSearchResult(true);
  
      const cancelToken = cancelTokenRef?.current?.token;
      const data = await DiscoveryRepo.fetchDiscoveryDataUser(searchText, isAnon, {cancelToken});
      if (data.success) {
        // ...
      }
-   } catch (error) {
-     setIsLoadingSearchResult(false);
-   }
+   } catch (error) {
+     // Handle error...
+   } finally {
+     setIsLoadingSearchResult(false);
+   }
  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->